provider plugin panic when trying to add multiple security policies

###Nutanix Cluster Information
Nutanix Cluster (Prism Element / AOS): 7.0.1
Nutanix Prism Central: pc.2024.3.1

###Terraform Version
Terraform v1.11.4
on darwin_arm64
provider registry.terraform.io/nutanix/nutanix v2.2.0

### Affected Resource(s)
* nutanix_network_security_policy_v2

### Terraform Configuration Files
variables.tf
```hcl
#define the type of variables to be used in terraform file
variable "nutanix_username" {
  type = string
}
variable "nutanix_password" {
  type = string
}
variable "nutanix_endpoint" {
  type = string
}
variable "vpc_name" {
  type = string
}
```
vpcs.tf:
```hcl
data "nutanix_vpcs_v2" "vpcs"{
  limit = 1
  filter = "name eq '${var.vpc_name}'"
}
````
main.tf:
```hcl
terraform {
  required_providers {
    nutanix = {
      source  = "nutanix/nutanix"
      version = ">= 2.0.0"
    }
  }
}

#defining nutanix configuration
provider "nutanix" {
  username = var.nutanix_username
  password = var.nutanix_password
  endpoint = var.nutanix_endpoint
  port     = 9440
  insecure = true
}
````
categories.tf:
```hcl
locals {
  app_count = 1
}


data "nutanix_categories_v2" "app_type" {
  count  = local.app_count
  limit  = 1
  filter = "key eq 'AppType' and value eq 'NVDMultiVmApp${format("%02d", count.index + 1)}'"
}

data "nutanix_categories_v2" "app_tier_app" {
  limit  = 1
  filter = "key eq 'AppTier' and value eq 'App'"
}

data "nutanix_categories_v2" "app_tier_db" {
  limit  = 1
  filter = "key eq 'AppTier' and value eq 'DB'"
}

data "nutanix_categories_v2" "app_tier_web" {
  limit  = 1
  filter = "key eq 'AppTier' and value eq 'Web'"
}
````
security_policies.tf:
```hcl
locals {
  isolation_policy_count = 1
  vlan_policies = 0
  vpc_policies = 2
}

resource "nutanix_network_security_policy_v2" "vlan_app" {
  count       = local.vlan_policies
  name        = "NVDMultiVmApp${format("%02d", count.index + 1)}"
  description = "Application policy for NVDMultiVmApp${format("%02d", count.index + 1)}"
  type        = "APPLICATION"
  state       = "ENFORCE"

  lifecycle {
    create_before_destroy = true
    ignore_changes = [name, description]
  }

#* outbound rules
  rules {
    description = "outbound for app tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_app.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = true
        src_allow_spec = "NONE"
      }
    }
  }

  rules {
    description = "outbound for db tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_db.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = true
        src_allow_spec = "NONE"
      }
    }
  }

  rules {
    description = "outbound for web tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_web.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = true
        src_allow_spec = "NONE"
      }
    }
  }

#* preventing vms with same tier from talking with each other
  rules {
    description = "deny amongst app tier"
    type        = "INTRA_GROUP"
    spec {
      intra_entity_group_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_app.categories[0].ext_id,
        ]
        secured_group_action = "DENY"
      }
    }
  }
  
  rules {
    description = "deny amongst db tier"
    type        = "INTRA_GROUP"
    spec {
      intra_entity_group_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_db.categories[0].ext_id,
        ]
        secured_group_action = "DENY"
      }
    }
  }

  rules {
    description = "deny amongst web tier"
    type        = "INTRA_GROUP"
    spec {
      intra_entity_group_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_web.categories[0].ext_id,
        ]
        secured_group_action = "DENY"
      }
    }
  }

#* ALL inbound rules
  rules {
    description = "ALL inbound for app tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_app.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = false
        src_allow_spec = "ALL"
        tcp_services {
          start_port = 22
          end_port = 22
        }
        tcp_services {
          start_port = 443
          end_port = 443
        }
        tcp_services {
          start_port = 2074
          end_port   = 2074
        }
        tcp_services {
          start_port = 3389
          end_port = 3389
        }
        tcp_services {
          start_port = 5985
          end_port   = 5985
        }
        icmp_services {
          type = 8
          code = 0
        }
      }
    }
  }

  rules {
    description = "ALL inbound for db tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_db.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = false
        src_allow_spec = "ALL"
        tcp_services {
          start_port = 22
          end_port = 22
        }
        tcp_services {
          start_port = 2074
          end_port   = 2074
        }
        tcp_services {
          start_port = 3389
          end_port = 3389
        }
        tcp_services {
          start_port = 5985
          end_port   = 5985
        }
        icmp_services {
          type = 8
          code = 0
        }
      }
    }
  }

  rules {
    description = "ALL inbound for web tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_web.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = false
        src_allow_spec = "ALL"
        tcp_services {
          start_port = 22
          end_port = 22
        }
        tcp_services {
          start_port = 443
          end_port = 443
        }
        tcp_services {
          start_port = 2074
          end_port   = 2074
        }
        tcp_services {
          start_port = 3389
          end_port = 3389
        }
        tcp_services {
          start_port = 5985
          end_port   = 5985
        }
        icmp_services {
          type = 8
          code = 0
        }
      }
    }
  }

#* inbound between tiers
  rules {
    description = "web inbound to app tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_app.categories[0].ext_id,
        ]
        src_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_web.categories[0].ext_id,
        ]
        is_all_protocol_allowed = false
        tcp_services {
          start_port = 2066
          end_port = 2066
        }
      }
    }
  }

  rules {
    description = "app inbound to db tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_db.categories[0].ext_id,
        ]
        src_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_app.categories[0].ext_id,
        ]
        is_all_protocol_allowed = false
        tcp_services {
          start_port = 3336
          end_port = 3336
        }
      }
    }
  }


  is_hitlog_enabled = true
}

resource "nutanix_network_security_policy_v2" "vpc_app" {
  count       = local.vpc_policies
  name        = "NVDMultiVmApp${format("%02d", count.index + 1)}"
  description = "Application policy for NVDMultiVmApp${format("%02d", count.index + 1)}"
  type        = "APPLICATION"
  state       = "ENFORCE"
  scope       = "VPC_LIST"
  
  vpc_reference = [
    data.nutanix_vpcs_v2.vpcs.vpcs[0].ext_id,
  ]

  lifecycle {
    create_before_destroy = true
    ignore_changes = [name, description]
  }

#* outbound rules
  rules {
    description = "outbound for app tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_app.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = true
        src_allow_spec = "NONE"
      }
    }
  }

  rules {
    description = "outbound for db tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_db.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = true
        src_allow_spec = "NONE"
      }
    }
  }

  rules {
    description = "outbound for web tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_web.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = true
        src_allow_spec = "NONE"
      }
    }
  }

#* preventing vms with same tier from talking with each other
  rules {
    description = "deny amongst app tier"
    type        = "INTRA_GROUP"
    spec {
      intra_entity_group_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_app.categories[0].ext_id,
        ]
        secured_group_action = "DENY"
      }
    }
  }
  
  rules {
    description = "deny amongst db tier"
    type        = "INTRA_GROUP"
    spec {
      intra_entity_group_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_db.categories[0].ext_id,
        ]
        secured_group_action = "DENY"
      }
    }
  }

  rules {
    description = "deny amongst web tier"
    type        = "INTRA_GROUP"
    spec {
      intra_entity_group_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_web.categories[0].ext_id,
        ]
        secured_group_action = "DENY"
      }
    }
  }

#* ALL inbound rules
  rules {
    description = "ALL inbound for app tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_app.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = false
        src_allow_spec = "ALL"
        tcp_services {
          start_port = 22
          end_port = 22
        }
        tcp_services {
          start_port = 443
          end_port = 443
        }
        tcp_services {
          start_port = 2074
          end_port   = 2074
        }
        tcp_services {
          start_port = 3389
          end_port = 3389
        }
        tcp_services {
          start_port = 5985
          end_port   = 5985
        }
        icmp_services {
          type = 8
          code = 0
        }
      }
    }
  }

  rules {
    description = "ALL inbound for db tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_db.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = false
        src_allow_spec = "ALL"
        tcp_services {
          start_port = 22
          end_port = 22
        }
        tcp_services {
          start_port = 2074
          end_port   = 2074
        }
        tcp_services {
          start_port = 3389
          end_port = 3389
        }
        tcp_services {
          start_port = 5985
          end_port   = 5985
        }
        icmp_services {
          type = 8
          code = 0
        }
      }
    }
  }

  rules {
    description = "ALL inbound for web tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_web.categories[0].ext_id,
        ]
        src_category_references = []
        is_all_protocol_allowed = false
        src_allow_spec = "ALL"
        tcp_services {
          start_port = 22
          end_port = 22
        }
        tcp_services {
          start_port = 443
          end_port = 443
        }
        tcp_services {
          start_port = 2074
          end_port   = 2074
        }
        tcp_services {
          start_port = 3389
          end_port = 3389
        }
        tcp_services {
          start_port = 5985
          end_port   = 5985
        }
        icmp_services {
          type = 8
          code = 0
        }
      }
    }
  }

#* inbound between tiers
  rules {
    description = "web inbound to app tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_app.categories[0].ext_id,
        ]
        src_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_web.categories[0].ext_id,
        ]
        is_all_protocol_allowed = false
        tcp_services {
          start_port = 2066
          end_port = 2066
        }
      }
    }
  }

  rules {
    description = "app inbound to db tier"
    type        = "APPLICATION"
    spec {
      application_rule_spec {
        secured_group_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_db.categories[0].ext_id,
        ]
        src_category_references = [
          try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
          data.nutanix_categories_v2.app_tier_app.categories[0].ext_id,
        ]
        is_all_protocol_allowed = false
        tcp_services {
          start_port = 3336
          end_port = 3336
        }
      }
    }
  }


  is_hitlog_enabled = true
}

resource "nutanix_network_security_policy_v2" "isolation" {
  count       = local.isolation_policy_count
  name        = "NVDMultiVmApp${format("%02d", count.index + 1)}_isolation"
  description = "Isolation rule for NVDMultiVmApp${format("%02d", count.index + 1)}"
  type        = "ISOLATION"
  state       = "ENFORCE"

  lifecycle {
    create_before_destroy = true
    ignore_changes = [name, description]
  }

  rules {
    description = "Isolation rule for NVDMultiVmApp${format("%02d", count.index + 1)}"
    type        = "MULTI_ENV_ISOLATION"
    spec {
      multi_env_isolation_rule_spec {
        spec {
          all_to_all_isolation_group {
            isolation_group {
              group_category_references = [
                try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
                data.nutanix_categories_v2.app_tier_app.categories[0].ext_id
              ]
            }
            isolation_group {
              group_category_references = [
                try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
                data.nutanix_categories_v2.app_tier_db.categories[0].ext_id
              ]
            }
            isolation_group {
              group_category_references = [
                try(data.nutanix_categories_v2.app_type[count.index].categories[0].ext_id, ""),
                data.nutanix_categories_v2.app_tier_web.categories[0].ext_id
              ]
            }
          }
        }
      }
    }
  }

  is_hitlog_enabled = true
}
```

### Panic Output
```sh
Error: Plugin did not respond
│ 
│   with nutanix_network_security_policy_v2.vpc_app[1],
│   on security_policies.tf line 277, in resource "nutanix_network_security_policy_v2" "vpc_app":
│  277: resource "nutanix_network_security_policy_v2" "vpc_app" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-nutanix_v2.2.0 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 47 [running]:
github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2.expandStringList(...)
        github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2/resource_nutanix_subnets_v2.go:1365
github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2.expandOneOfNetworkSecurityPolicyRuleSpec({0x10376be80?, 0x14000b8ec48?})
        github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2.go:809 +0x1c9c
github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2.expandNetworkSecurityPolicyRule({0x1400034d1e0, 0xb, 0x14000e02ca0?})
        github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2.go:772 +0x3cc
github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2.ResourceNutanixNetworkSecurityPolicyV2Create({0x10399aeb8, 0x140014957a0}, 0x0?, {0x10374bf40?, 0x14000c83440?})
        github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2.go:455 +0x568
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x140009a7880, {0x10399ae48, 0x1400028ed20}, 0x2?, {0x10374bf40, 0x14000c83440})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:341 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x140009a7880, {0x10399ae48, 0x1400028ed20}, 0x1400116edd0, 0x14000fb7280, {0x10374bf40, 0x14000c83440})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:467 +0x67c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400024e078, {0x10399ae48, 0x1400028ed20}, 0x1400028ed70)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:977 +0xabc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14000116400, {0x10399ae10?, 0x14000d981e0?}, 0x14000ed4000)
        github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x20c
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x10393e060?, 0x14000116400}, {0x10399ae10, 0x14000d981e0}, 0x14000b905a0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002df500, {0x10399f8d0, 0x140001ee780}, 0x14000f02000, 0x140007d4d20, 0x104136ea0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0xa58
google.golang.org/grpc.(*Server).handleStream(0x140002df500, {0x10399f8d0, 0x140001ee780}, 0x14000f02000, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0x7cc
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.32.0/server.go:859 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 39
        google.golang.org/grpc@v1.32.0/server.go:857 +0x24c

Error: The terraform-provider-nutanix_v2.2.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behavior
2 application security and 1 isolation policies should be created.

### Actual Behavior
provider plugin crashes and no policies are created.

### Steps to Reproduce
1. `terraform apply`
2. if you change vpc policies to 1, it works.

adding relevant trace log for provider:
```sh
2025-04-30T09:00:58.140+0200 [TRACE] provider.terraform-provider-nutanix_v2.2.0: Served request: tf_req_id=5c3e0c7c-9dbb-a994-78d8-24c6934c0046 tf_proto_version=5 tf_provider_addr=provider tf_rpc=ApplyResourceChange @caller=runtime/panic.go:914 @module=sdk.proto tf_resource_type=nutanix_network_security_policy_v2 timestamp="2025-04-30T09:00:58.140+0200"
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: panic: interface conversion: interface {} is nil, not string
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: goroutine 16 [running]:
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2.expandStringList(...)
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2/resource_nutanix_subnets_v2.go:1365
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2.expandOneOfNetworkSecurityPolicyRuleSpec({0x105243e80?, 0x1400153ae70?})
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2.go:809 +0x1c9c
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2.expandNetworkSecurityPolicyRule({0x14000fda160, 0xb, 0x14000c4d1b0?})
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2.go:772 +0x3cc
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2.ResourceNutanixNetworkSecurityPolicyV2Create({0x105472eb8, 0x140002abb90}, 0x0?, {0x105223f40?, 0x1400164e510?})
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2.go:455 +0x568
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x14000b3f880, {0x105472e48, 0x140014f6dc0}, 0x2?, {0x105223f40, 0x1400164e510})
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:341 +0xe8
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000b3f880, {0x105472e48, 0x140014f6dc0}, 0x14001513450, 0x14001651b00, {0x105223f40, 0x1400164e510})
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:467 +0x67c
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400024e090, {0x105472e48, 0x140014f6dc0}, 0x140014f6e10)
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:977 +0xabc
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14000116400, {0x105472e10?, 0x14000f306f0?}, 0x140014e18f0)
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x20c
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x105416060?, 0x14000116400}, {0x105472e10, 0x14000f306f0}, 0x140013bb920, 0x0)
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x164
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: google.golang.org/grpc.(*Server).processUnaryRPC(0x140002df500, {0x1054778d0, 0x140001ee780}, 0x14000f02200, 0x14000a3f1a0, 0x105c0eea0, 0x0)
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	google.golang.org/grpc@v1.32.0/server.go:1194 +0xa58
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: google.golang.org/grpc.(*Server).handleStream(0x140002df500, {0x1054778d0, 0x140001ee780}, 0x14000f02200, 0x0)
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	google.golang.org/grpc@v1.32.0/server.go:1517 +0x7cc
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: google.golang.org/grpc.(*Server).serveStreams.func1.2()
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	google.golang.org/grpc@v1.32.0/server.go:859 +0x84
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 24
2025-04-30T09:00:58.143+0200 [DEBUG] provider.terraform-provider-nutanix_v2.2.0: 	google.golang.org/grpc@v1.32.0/server.go:857 +0x24c
2025-04-30T09:00:58.144+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-04-30T09:00:58.144+0200 [ERROR] provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/nutanix/nutanix/2.2.0/darwin_arm64/terraform-provider-nutanix_v2.2.0 id=23869 error="exit status 2"
2025-04-30T09:00:58.156+0200 [DEBUG] provider: plugin exited
```

--- a/nutanix/common/common.go
+++ b/nutanix/common/common.go
@@ -1,9 +1,7 @@
 package common
 
 import (
-	"fmt"
 	"log"
-	"regexp"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
@@ -38,16 +36,4 @@ func IsExplicitlySet(d *schema.ResourceData, key string) bool {
 		return !val.IsNull() // Ensure key exists and isn't explicitly null
 	}
 	return false
-}
-
-// ExtIdValidation checks if a given string is a valid UUID.
-// It uses a regular expression to validate the UUID format.
-var uuidRegex = regexp.MustCompile(`^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$`)
-
-func ExtIDValidation(val interface{}, key string) (warns []string, errs []error) {
-	v := val.(string)
-	if !uuidRegex.MatchString(v) {
-		errs = append(errs, fmt.Errorf("%q must be a valid Ext ID: got %q", key, v))
-	}
-	return
 }

--- a/nutanix/common/common.go
+++ b/nutanix/common/common.go
@@ -44,7 +44,7 @@ func IsExplicitlySet(d *schema.ResourceData, key string) bool {
 // It uses a regular expression to validate the UUID format.
 var uuidRegex = regexp.MustCompile(`^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$`)
 
-func ExtIdValidation(val interface{}, key string) (warns []string, errs []error) {
+func ExtIDValidation(val interface{}, key string) (warns []string, errs []error) {
 	v := val.(string)
 	if !uuidRegex.MatchString(v) {
 		errs = append(errs, fmt.Errorf("%q must be a valid Ext ID: got %q", key, v))

--- a/nutanix/common/common.go
+++ b/nutanix/common/common.go
@@ -12,13 +12,13 @@ import (
 // which is defined in the schema
 // its return a list of string
 func ExpandListOfString(list []interface{}) []string {
-	stringListStr := make([]string, len(list))
+	stringListStr := make([]string, 0)
 	for i, v := range list {
 		if v == nil || v == "" {
 			log.Printf("[DEBUG] Skipping nil or empty value at index %d", i)
 			continue // Skip nil or empty values
 		}
-		stringListStr[i] = v.(string)
+		stringListStr = append(stringListStr, v.(string))
 	}
 	return stringListStr
 }

--- a/nutanix/common/common.go
+++ b/nutanix/common/common.go
@@ -1,7 +1,9 @@
 package common
 
 import (
+	"fmt"
 	"log"
+	"regexp"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
@@ -12,6 +14,10 @@ import (
 func ExpandListOfString(list []interface{}) []string {
 	stringListStr := make([]string, len(list))
 	for i, v := range list {
+		if v == nil || v == "" {
+			log.Printf("[DEBUG] Skipping nil or empty value at index %d", i)
+			continue // Skip nil or empty values
+		}
 		stringListStr[i] = v.(string)
 	}
 	return stringListStr
@@ -32,4 +38,16 @@ func IsExplicitlySet(d *schema.ResourceData, key string) bool {
 		return !val.IsNull() // Ensure key exists and isn't explicitly null
 	}
 	return false
+}
+
+// ExtIdValidation checks if a given string is a valid UUID.
+// It uses a regular expression to validate the UUID format.
+var uuidRegex = regexp.MustCompile(`^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$`)
+
+func ExtIdValidation(val interface{}, key string) (warns []string, errs []error) {
+	v := val.(string)
+	if !uuidRegex.MatchString(v) {
+		errs = append(errs, fmt.Errorf("%q must be a valid Ext ID: got %q", key, v))
+	}
+	return
 }

--- a/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2.go
+++ b/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2.go
@@ -14,6 +14,7 @@ import (
 	import4 "github.com/nutanix/ntnx-api-golang-clients/microseg-go-client/v4/models/prism/v4/config"
 	import2 "github.com/nutanix/ntnx-api-golang-clients/prism-go-client/v4/models/prism/v4/config"
 	conns "github.com/terraform-providers/terraform-provider-nutanix/nutanix"
+	"github.com/terraform-providers/terraform-provider-nutanix/nutanix/common"
 	"github.com/terraform-providers/terraform-provider-nutanix/utils"
 )
 
@@ -77,14 +78,16 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Type:     schema.TypeList,
 													Required: true,
 													Elem: &schema.Schema{
-														Type: schema.TypeString,
+														Type:         schema.TypeString,
+														ValidateFunc: common.ExtIdValidation,
 													},
 												},
 												"second_isolation_group": {
 													Type:     schema.TypeList,
 													Required: true,
 													Elem: &schema.Schema{
-														Type: schema.TypeString,
+														Type:         schema.TypeString,
+														ValidateFunc: common.ExtIdValidation,
 													},
 												},
 											},
@@ -99,7 +102,8 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Type:     schema.TypeList,
 													Required: true,
 													Elem: &schema.Schema{
-														Type: schema.TypeString,
+														Type:         schema.TypeString,
+														ValidateFunc: common.ExtIdValidation,
 													},
 												},
 												"src_allow_spec": {
@@ -127,7 +131,8 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Optional: true,
 													Computed: true,
 													Elem: &schema.Schema{
-														Type: schema.TypeString,
+														Type:         schema.TypeString,
+														ValidateFunc: common.ExtIdValidation,
 													},
 												},
 												"src_subnet": {
@@ -173,7 +178,8 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Optional: true,
 													Computed: true,
 													Elem: &schema.Schema{
-														Type: schema.TypeString,
+														Type:         schema.TypeString,
+														ValidateFunc: common.ExtIdValidation,
 													},
 												},
 												"dest_address_group_references": {
@@ -181,7 +187,8 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Optional: true,
 													Computed: true,
 													Elem: &schema.Schema{
-														Type: schema.TypeString,
+														Type:         schema.TypeString,
+														ValidateFunc: common.ExtIdValidation,
 													},
 												},
 												"service_group_references": {
@@ -189,7 +196,8 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Optional: true,
 													Computed: true,
 													Elem: &schema.Schema{
-														Type: schema.TypeString,
+														Type:         schema.TypeString,
+														ValidateFunc: common.ExtIdValidation,
 													},
 												},
 												"is_all_protocol_allowed": {
@@ -276,7 +284,8 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Computed: true,
 													Optional: true,
 													Elem: &schema.Schema{
-														Type: schema.TypeString,
+														Type:         schema.TypeString,
+														ValidateFunc: common.ExtIdValidation,
 													},
 												},
 											},
@@ -309,7 +318,8 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 																						Type:     schema.TypeList,
 																						Required: true,
 																						Elem: &schema.Schema{
-																							Type: schema.TypeString,
+																							Type:         schema.TypeString,
+																							ValidateFunc: common.ExtIdValidation,
 																						},
 																					},
 																				},
@@ -472,7 +482,7 @@ func ResourceNutanixNetworkSecurityPolicyV2Create(ctx context.Context, d *schema
 		spec.Scope = &p
 	}
 	if vpcRef, ok := d.GetOk("vpc_reference"); ok {
-		spec.VpcReferences = expandStringList(vpcRef.([]interface{}))
+		spec.VpcReferences = common.ExpandListOfString(vpcRef.([]interface{}))
 	}
 
 	aJSON, _ := json.MarshalIndent(spec, "", "  ")
@@ -685,7 +695,7 @@ func ResourceNutanixNetworkSecurityPolicyV2Update(ctx context.Context, d *schema
 		updatedSpec.Scope = &p
 	}
 	if d.HasChange("vpc_reference") {
-		updatedSpec.VpcReferences = expandStringList(d.Get("vpc_reference").([]interface{}))
+		updatedSpec.VpcReferences = common.ExpandListOfString(d.Get("vpc_reference").([]interface{}))
 	}
 
 	aJSON, _ := json.MarshalIndent(updatedSpec, "", "  ")
@@ -791,10 +801,10 @@ func expandOneOfNetworkSecurityPolicyRuleSpec(pr interface{}) *import1.OneOfNetw
 			isoVal := isoI[0].(map[string]interface{})
 
 			if firstIso, ok := isoVal["first_isolation_group"]; ok && len(firstIso.([]interface{})) > 0 {
-				iso.FirstIsolationGroup = expandStringList(firstIso.([]interface{}))
+				iso.FirstIsolationGroup = common.ExpandListOfString(firstIso.([]interface{}))
 			}
 			if secIso, ok := isoVal["second_isolation_group"]; ok && len(secIso.([]interface{})) > 0 {
-				iso.SecondIsolationGroup = expandStringList(secIso.([]interface{}))
+				iso.SecondIsolationGroup = common.ExpandListOfString(secIso.([]interface{}))
 			}
 			policyRules.SetValue(*iso)
 		}
@@ -806,7 +816,7 @@ func expandOneOfNetworkSecurityPolicyRuleSpec(pr interface{}) *import1.OneOfNetw
 			appVal := appI[0].(map[string]interface{})
 
 			if secGroup, ok := appVal["secured_group_category_references"]; ok && len(secGroup.([]interface{})) > 0 {
-				app.SecuredGroupCategoryReferences = expandStringList(secGroup.([]interface{}))
+				app.SecuredGroupCategoryReferences = common.ExpandListOfString(secGroup.([]interface{}))
 			}
 			if srcAllow, ok := appVal["src_allow_spec"]; ok && len(srcAllow.(string)) > 0 {
 				const two, three = 2, 3
@@ -829,10 +839,10 @@ func expandOneOfNetworkSecurityPolicyRuleSpec(pr interface{}) *import1.OneOfNetw
 				app.DestAllowSpec = &p
 			}
 			if srcCatRef, ok := appVal["src_category_references"]; ok && len(srcCatRef.([]interface{})) > 0 {
-				app.SrcCategoryReferences = expandStringList(srcCatRef.([]interface{}))
+				app.SrcCategoryReferences = common.ExpandListOfString(srcCatRef.([]interface{}))
 			}
 			if destCatRef, ok := appVal["dest_category_references"]; ok && len(destCatRef.([]interface{})) > 0 {
-				app.DestCategoryReferences = expandStringList(destCatRef.([]interface{}))
+				app.DestCategoryReferences = common.ExpandListOfString(destCatRef.([]interface{}))
 			}
 			if srcSubnet, ok := appVal["src_subnet"]; ok && len(srcSubnet.([]interface{})) > 0 {
 				app.SrcSubnet = expandIPv4AddressMicroseg(srcSubnet)
@@ -841,13 +851,13 @@ func expandOneOfNetworkSecurityPolicyRuleSpec(pr interface{}) *import1.OneOfNetw
 				app.DestSubnet = expandIPv4AddressMicroseg(destSubnet)
 			}
 			if srcAddGrpRef, ok := appVal["src_address_group_references"]; ok && len(srcAddGrpRef.([]interface{})) > 0 {
-				app.SrcAddressGroupReferences = expandStringList(srcAddGrpRef.([]interface{}))
+				app.SrcAddressGroupReferences = common.ExpandListOfString(srcAddGrpRef.([]interface{}))
 			}
 			if destAddGrpRef, ok := appVal["dest_address_group_references"]; ok && len(destAddGrpRef.([]interface{})) > 0 {
-				app.DestAddressGroupReferences = expandStringList(destAddGrpRef.([]interface{}))
+				app.DestAddressGroupReferences = common.ExpandListOfString(destAddGrpRef.([]interface{}))
 			}
 			if serviceGrpRef, ok := appVal["service_group_references"]; ok && len(serviceGrpRef.([]interface{})) > 0 {
-				app.ServiceGroupReferences = expandStringList(serviceGrpRef.([]interface{}))
+				app.ServiceGroupReferences = common.ExpandListOfString(serviceGrpRef.([]interface{}))
 			}
 			if allProto, ok := appVal["is_all_protocol_allowed"]; ok {
 				app.IsAllProtocolAllowed = utils.BoolPtr(allProto.(bool))
@@ -875,7 +885,7 @@ func expandOneOfNetworkSecurityPolicyRuleSpec(pr interface{}) *import1.OneOfNetw
 			intraVal := intraI[0].(map[string]interface{})
 
 			if secGroup, ok := intraVal["secured_group_category_references"]; ok && len(secGroup.([]interface{})) > 0 {
-				intra.SecuredGroupCategoryReferences = expandStringList(secGroup.([]interface{}))
+				intra.SecuredGroupCategoryReferences = common.ExpandListOfString(secGroup.([]interface{}))
 			}
 			if secGroupAction, ok := intraVal["secured_group_action"]; ok && len(secGroupAction.(string)) > 0 {
 				const two, three = 2, 3
@@ -961,7 +971,7 @@ func expandIsolationGroup(isolationGroup []interface{}) []import1.IsolationGroup
 			iso := import1.IsolationGroup{}
 
 			if groupCat, ok := val["group_category_references"]; ok && len(groupCat.([]interface{})) > 0 {
-				iso.GroupCategoryReferences = expandStringList(groupCat.([]interface{}))
+				iso.GroupCategoryReferences = common.ExpandListOfString(groupCat.([]interface{}))
 			}
 			isolations[k] = iso
 		}

--- a/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2.go
+++ b/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2.go
@@ -78,16 +78,14 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Type:     schema.TypeList,
 													Required: true,
 													Elem: &schema.Schema{
-														Type:         schema.TypeString,
-														ValidateFunc: common.ExtIdValidation,
+														Type: schema.TypeString,
 													},
 												},
 												"second_isolation_group": {
 													Type:     schema.TypeList,
 													Required: true,
 													Elem: &schema.Schema{
-														Type:         schema.TypeString,
-														ValidateFunc: common.ExtIdValidation,
+														Type: schema.TypeString,
 													},
 												},
 											},
@@ -102,8 +100,7 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Type:     schema.TypeList,
 													Required: true,
 													Elem: &schema.Schema{
-														Type:         schema.TypeString,
-														ValidateFunc: common.ExtIdValidation,
+														Type: schema.TypeString,
 													},
 												},
 												"src_allow_spec": {
@@ -131,8 +128,7 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Optional: true,
 													Computed: true,
 													Elem: &schema.Schema{
-														Type:         schema.TypeString,
-														ValidateFunc: common.ExtIdValidation,
+														Type: schema.TypeString,
 													},
 												},
 												"src_subnet": {
@@ -178,8 +174,7 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Optional: true,
 													Computed: true,
 													Elem: &schema.Schema{
-														Type:         schema.TypeString,
-														ValidateFunc: common.ExtIdValidation,
+														Type: schema.TypeString,
 													},
 												},
 												"dest_address_group_references": {
@@ -187,8 +182,7 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Optional: true,
 													Computed: true,
 													Elem: &schema.Schema{
-														Type:         schema.TypeString,
-														ValidateFunc: common.ExtIdValidation,
+														Type: schema.TypeString,
 													},
 												},
 												"service_group_references": {
@@ -196,8 +190,7 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Optional: true,
 													Computed: true,
 													Elem: &schema.Schema{
-														Type:         schema.TypeString,
-														ValidateFunc: common.ExtIdValidation,
+														Type: schema.TypeString,
 													},
 												},
 												"is_all_protocol_allowed": {
@@ -284,8 +277,7 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 													Computed: true,
 													Optional: true,
 													Elem: &schema.Schema{
-														Type:         schema.TypeString,
-														ValidateFunc: common.ExtIdValidation,
+														Type: schema.TypeString,
 													},
 												},
 											},
@@ -318,8 +310,7 @@ func ResourceNutanixNetworkSecurityPolicyV2() *schema.Resource {
 																						Type:     schema.TypeList,
 																						Required: true,
 																						Elem: &schema.Schema{
-																							Type:         schema.TypeString,
-																							ValidateFunc: common.ExtIdValidation,
+																							Type: schema.TypeString,
 																						},
 																					},
 																				},

--- a/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2_test.go
+++ b/nutanix/services/networkingv2/resource_nutanix_network_security_policies_v2_test.go
@@ -2,6 +2,7 @@ package networkingv2_test
 
 import (
 	"fmt"
+	"regexp"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
@@ -89,6 +90,24 @@ func TestAccV2NutanixNetworkSecurityResource_WithMultiEnvIsolationRuleSpecRule(t
 					resource.TestCheckResourceAttr(resourceNameNs, "rules.0.type", "MULTI_ENV_ISOLATION"),
 					resource.TestCheckResourceAttrSet(resourceNameNs, "vpc_reference.#"),
 					resource.TestCheckResourceAttr(resourceNameNs, "is_hitlog_enabled", "false"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccV2NutanixNetworkSecurityResource_InvalidExtIDReference(t *testing.T) {
+	r := acctest.RandInt()
+	name := fmt.Sprintf("tf-test-nsp-%d", r)
+	desc := "test nsp description"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { acc.TestAccPreCheck(t) },
+		Providers: acc.TestAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testNetworkSecurityInvalidConfig(name, desc),
+				ExpectError: regexp.MustCompile(
+					`(?s)must be a valid Ext ID: got "".*must be a valid Ext ID: got "invalid-ext-id"`,
 				),
 			},
 		},
@@ -334,4 +353,29 @@ func testNetworkSecurityConfigWithMultiEnvIsolationRuleSpecRule(name, desc strin
 		depends_on = [nutanix_vpc_v2.test]
 	  }
 	  `, name, desc)
+}
+
+func testNetworkSecurityInvalidConfig(name, desc string) string {
+	return fmt.Sprintf(`
+	resource "nutanix_network_security_policy_v2" "test" {
+		name = "%[1]s"
+		description = "%[2]s"
+		state = "SAVE"
+		type = "ISOLATION"
+		rules{
+		  type = "TWO_ENV_ISOLATION"
+		  spec{
+			two_env_isolation_rule_spec{
+			  first_isolation_group = [
+				"",
+			  ]
+			  second_isolation_group =  [
+				"invalid-ext-id",
+			  ]
+			}
+		  }
+		}
+		is_hitlog_enabled = true
+	  }
+`, name, desc)
 }

--- a/nutanix/services/networkingv2/resource_nutanix_subnets_v2.go
+++ b/nutanix/services/networkingv2/resource_nutanix_subnets_v2.go
@@ -846,7 +846,7 @@ func expandDhcpOptions(pr []interface{}) *import1.DhcpOptions {
 			dhcpOps.DomainName = utils.StringPtr(dn.(string))
 		}
 		if searchDomain, ok := val["search_domains"]; ok && len(searchDomain.([]interface{})) > 0 {
-			dhcpOps.SearchDomains = expandStringList(searchDomain.([]interface{}))
+			dhcpOps.SearchDomains = common.ExpandListOfString(searchDomain.([]interface{}))
 		}
 		if tftp, ok := val["tftp_server_name"]; ok && len(tftp.(string)) > 0 {
 			dhcpOps.TftpServerName = utils.StringPtr(tftp.(string))
@@ -1117,7 +1117,7 @@ func expandExternalSubnet(pr []interface{}) []import1.ExternalSubnet {
 				sub.ExternalIps = expandIPAddress(extips.([]interface{}))
 			}
 			if gatewayNodes, ok := val["gateway_nodes"]; ok && len(gatewayNodes.([]interface{})) > 0 {
-				sub.GatewayNodes = expandStringList(gatewayNodes.([]interface{}))
+				sub.GatewayNodes = common.ExpandListOfString(gatewayNodes.([]interface{}))
 			}
 			if activeGatewayNode, ok := val["active_gateway_node"]; ok && len(activeGatewayNode.([]interface{})) > 0 {
 				sub.ActiveGatewayNodes = expandGatewayNodeReference(activeGatewayNode)
@@ -1353,18 +1353,6 @@ func expandIPv6Pool(pr []interface{}) []import1.IPv6Pool {
 			pools[k] = pool
 		}
 		return pools
-	}
-	return nil
-}
-
-func expandStringList(pr []interface{}) []string {
-	if len(pr) > 0 {
-		strList := make([]string, len(pr))
-
-		for k, v := range pr {
-			strList[k] = v.(string)
-		}
-		return strList
 	}
 	return nil
 }

